agent/command/expansion_update.go

```diff
@@ -119,13 +136,18 @@ func (c *update) Execute(ctx context.Context,
 		}
 
 		logger.Task().Infof("Updating expansions with keys from file '%s'.", filename)
-		err := conf.NewExpansions.UpdateFromYaml(filename)
+		keys, err := conf.NewExpansions.UpdateFromYaml(filename)
 		if err != nil {
 			return errors.WithStack(err)
 		}
-		if err = conf.DynamicExpansions.UpdateFromYaml(filename); err != nil {
+		if _, err = conf.DynamicExpansions.UpdateFromYaml(filename); err != nil {
 			return errors.WithStack(err)
 		}
+		if c.RedactFileExpansions {
+			for _, key := range keys {
+				conf.NewExpansions.Redact(key)
```

Comment:
Not a huge deal, but re the above comment about the mutex lock, this isn't an atomic operation, so the task log could read the redacted expansion in plaintext in between L139-147. I'm alright if we just kind of leave it, but just awareness that it's not airtight.

Reply:
Not sure, but for this comment, did you decide to just keep the two operations non-atomic since it's a very small case? I don't see any specific change for this. If we wanted to fix the race, then `UpdateFromYaml` + `RedactKey` would have to hold the same lock on both operations because it's updating the keys and leaving them public before `RedactKey` fixes them up.